pysite/views/api/bot/doc.py

```diff
@@ -0,0 +1,110 @@
+from flask import jsonify
+from schema import Optional, Schema
+
+from pysite.base_route import APIView
+from pysite.constants import ValidationTypes
+from pysite.decorators import api_key, api_params
+from pysite.mixins import DBMixin
+
+
+GET_SCHEMA = Schema([
+    {
+        Optional("package"): str
+    }
+])
+
+POST_SCHEMA = Schema([
+    {
+        "package": str,
+        "base_url": str,
+        "inventory_url": str
+    }
+])
+
+DELETE_SCHEMA = Schema([
+    {
+        "package": str
+    }
+])
+
+
+class DocView(APIView, DBMixin):
+    path = "/docs"
+    name = "api.bot.docs"
+    table_name = "pydoc_links"
+
+    @api_key
+    @api_params(schema=GET_SCHEMA, validation_type=ValidationTypes.params)
+    def get(self, params=None):
+        """
+        Fetches documentation metadata from the database.
+
+        - If `package` is provided, fetch metadata
+        for the given package, or `{}` if not found.
+
+        - If `package` is not provided, return all
+        packages known to the database.
+
+        Data must be provided as params.
+        API key must be provided as header.
+        """
+
+        if params:
+            package = params[0].get("package")
```

Comment:
This schema setup is intended to allow one to specify multiple sets of parameters. You should support this in your endpoints.